### PR TITLE
Focused UI around GameLauncher and ServerLauncher test tool functiona…

### PR DESCRIPTION
**Pull request by AMZN-scspaldi**
…lity.

Signed-off-by: scspaldi <scspaldi@amazon.com>

### CODE REVIEW

**Review by Kadino — APPROVED** [review]
I approve this rename, and the deprecation pattern.

Interestingly, pytest has a CLI arg to fail-on-warning: https://docs.pytest.org/en/latest/how-to/capture-warnings.html

File: Tools/LyTestTools/ly_test_tools/launchers/launcher_helper.py [hunk 1]
```diff
@@ -8,11 +8,18 @@
...
 
 import ly_test_tools
 import ly_test_tools._internal.managers.workspace as workspace_manager
 import ly_test_tools.launchers.platforms.base as base_launcher
 
+
+# These are the launchers *currently* supported by the test tools. While other launchers exist, they are not supported
+# by the test tools.
+GAME_LAUNCHERS = ['windows', 'linux', 'android']
+SERVER_LAUNCHERS = ['linux', 'linux_dedicated']
```
Comment (AMZN-Gene) [inline Tools/LyTestTools/ly_test_tools/launchers/launcher_helper.py:21]: No windows server launchers? So if we want to run a Windows test we'll need 2 PCs? A Windows and LInux PC?

**Review by AMZN-Gene — APPROVED** [review]

File: Tools/LyTestTools/ly_test_tools/launchers/launcher_helper.py (hunk 1 above)
Comment (AMZN-scspaldi) [reply Tools/LyTestTools/ly_test_tools/launchers/launcher_helper.py:21]: Good catch, fixed.

File: Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/test_tools_fixtures.py [hunk 2]
```diff
@@ -262,10 +262,10 @@ def _dedicated_launcher(request, workspace, launcher_platform, level=""):
     """Separate implementation to call directly during unit tests"""
 
     if not level:
-        launcher = ly_test_tools.launchers.launcher_helper.create_dedicated_launcher(
+        launcher = ly_test_tools.launchers.launcher_helper.create_server_launcher(
```
[the remark this answers is not in the document]
Comment (AMZN-scspaldi) [reply Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/test_tools_fixtures.py:265]: Added a deprecation on it.

**Review by evanchia-ly-sdets — APPROVED** [review]

**Pull request merged**